packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -72,16 +72,65 @@ function run (ipc, configFile, projectRoot) {
     }
 
     ipc.send('setupTestingType:error', util.serializeError(
-      require('@packages/errors').getError('CONFIG_FILE_DEV_SERVER_IS_NOT_A_FUNCTION', configFile, config),
+      require('@packages/errors').getError('CONFIG_FILE_DEV_SERVER_IS_NOT_A_FUNCTION', file, config),
     ))
 
     return false
   }
 
+  ipc.on('loadLegacyPlugins', async (legacyConfig) => {
+    try {
+      let legacyPlugins = require(file)
+
+      if (legacyPlugins && typeof legacyPlugins.default === 'function') {
+        legacyPlugins = legacyPlugins.default
+      }
+
+      // invalid or empty plugins file
+      if (typeof legacyPlugins !== 'function') {
+        ipc.send('loadLegacyPlugins:reply', legacyConfig)
+
+        return
+      }
+
+      // we do not want to execute any tasks - the purpose
+      // of this is to get any modified config returned
+      // by plugins.
+      const noop = () => {}
+      const legacyPluginsConfig = await legacyPlugins(noop, legacyConfig)
+
+      // match merging strategy from 9.x
+      const mergedLegacyConfig = {
+        ...legacyConfig,
+        ...legacyPluginsConfig,
+      }
+
+      if (legacyConfig.e2e || legacyPluginsConfig.e2e) {
```

Comment:
If the user's plugin do not return `config` which I think can happen based on [this](https://docs.cypress.io/guides/tooling/plugins-guide#Using-a-plugin) - we are throwing an error, we should take into account when this happens and `legacyPluginsConfig` is `undefined`

<img width="956" alt="image" src="https://user-images.githubusercontent.com/15656860/157750641-c6a85630-d48d-47a9-b76e-e9806403d770.png">

Reply:
added test and fixed https://github.com/cypress-io/cypress/pull/20509/commits/5f11b6ba8c637d67b0efc608d0973f95234e47d3